pkg/cloudprovider/aws/object_storage_adapter.go

```diff
@@ -27,14 +27,16 @@ import (
 var _ cloudprovider.ObjectStorageAdapter = &objectStorageAdapter{}
 
 type objectStorageAdapter struct {
-	s3 *s3.S3
+	s3       *s3.S3
+	KMSKeyID string
 }
 
 func (op *objectStorageAdapter) PutObject(bucket string, key string, body io.ReadSeeker) error {
 	req := &s3.PutObjectInput{
-		Bucket: &bucket,
-		Key:    &key,
-		Body:   body,
+		Bucket:      &bucket,
+		Key:         &key,
+		SSEKMSKeyId: &op.KMSKeyID,
```

Comment:
Given that we need to set `ServerSideEncryption` based on the presence of `kmsKeyID`, let's only set `ServerSideEncryption` and `SSEKMSKeyId` if necessary.